File: floodFill/flood_fill.py
def floodFill(image, sr, sc, color):
    initial_color = image[sr][sc]
    fill(image, sr, sc, color, initial_color)

    return image

def fill(image, sr, sc, color, initial_color):
    if sr < 0 or sr > len(image) -1 or sc < 0 or sc > len(image[0]) -1 or image[sr][sc] != initial_color or image[sr][sc] == color:
        return
    
    image[sr][sc] = color

    fill(image, sr + 1, sc, color, initial_color)
    fill(image, sr - 1, sc, color, initial_color)
    fill(image, sr, sc + 1 , color, initial_color)
    fill(image, sr, sc - 1, color, initial_color)


print(floodFill([[1,1,1],[1,1,0],[1,0,1]], 1, 1, 2))

File: Week1/floodFill/floodFill.js
const floodFill = (image, sr, sc, newColor) => {
    const initialColor = image[sr][sc];
    dfs(image, sr, sc, newColor, initialColor);
    return image;
}


const dfs = (image, sr, sc, newColor, initialColor) => {
    if (sr < 0 || sr > image.length -1 || sc < 0 || sc > image[0].length -1 || image[sr][sc] != initialColor || image[sr][sc] == newColor) {
        return image;
    }

    image[sr][sc] = newColor;
    dfs(image, sr + 1, sc, newColor, initialColor);
    dfs(image, sr - 1, sc, newColor, initialColor);
    dfs(image, sr, sc + 1, newColor, initialColor);
    dfs(image, sr, sc - 1, newColor, initialColor);
}



console.log(floodFill([[1,1,1],[1,1,0],[1,0,1]], 1, 1, 2))